x/staking/types/msg.go

```diff
@@ -178,16 +178,22 @@ func (msg MsgEditValidator) ValidateBasic() sdk.Error {
 		return sdk.NewError(DefaultCodespace, CodeInvalidInput, "nil validator address")
 	}
 
+	// TODO what if the validator just wanted to change their commission rate, do they need a description
 	if msg.Description == (Description{}) {
 		return sdk.NewError(DefaultCodespace, CodeInvalidInput, "transaction must include some information to modify")
 	}
 
+	if msg.CommissionRate != nil {
+		if msg.CommissionRate.GT(sdk.OneDec()) || msg.CommissionRate.LT(sdk.ZeroDec()) {
+			return sdk.NewError(DefaultCodespace, CodeInvalidInput, "commission rate must be between 0 and 1, inclusive")
+		}
+	}
```

Comment:
Remove these checks, instead we should be calling 
`(c Commission) ValidateNewRate(newRate sdk.Dec, blockTime time.Time)`
 from `x/staking/commission.go`

OR add relevant logic to that file in a new function

Reply:
Hmm but what block time could be used?

Reply:
No need to. `Commission` already has a `Validate` method you can call here. Just use that.